### PR TITLE
migrate rails views to js templates

### PR DESCRIPTION
**Pull request by matallo**
re: #12667

this will eventually point to `master` once https://github.com/CartoDB/cartodb/pull/12739 is merged

---

### CODE REVIEW

File: lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  _onDataError: function (e) {
+    this._hideBlocks();
+    this._showBlocks([ 'filters', 'error' ]);
+  },
+
+  _showBlocks: function (views) {
+    var self = this;
+    if (views) {
+      _.each(views, function (v) {
+        self.controlledViews[v].show();
```
Comment (rubenmoya) [inline lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js]: I know you'll hate me, but `v` -> `view`

File: lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        v.show();
+        self.enabledViews.push(v);
+      });
+    }
+  },
+
+  _hideBlocks: function (views) {
+    var self = this;
+    if (views) {
+      _.each(views, function (v) {
```
Comment (rubenmoya) [inline lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js]: `v` -> `view`

File: lib/assets/javascripts/cartodb/dashboard/main_view_static.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  },
+
+  initialize: function () {
+    this._initModels();
+    this._initViews();
+    this._initBindings();
+  },
+
+  _initBindings: function () {
+    this.router.model.bind('change', this._onRouterChange, this);
```
Comment (rubenmoya) [inline lib/assets/javascripts/cartodb/dashboard/main_view_static.js]: Could we use `listenTo` here? I don't really know the difference, just asking.

File: lib/assets/javascripts/cartodb/common/support_view_static.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  _getUserType: function () {
+    var accountType = this.user.get('account_type').toLowerCase();
+
+    // Get user type
+    if (this.user.isOrgOwner()) {
+      return 'org_admin';
+    } else if (this.user.isInsideOrg()) {
+      return 'org';
+    } else if (accountType === 'internal' || accountType === 'partner' || accountType === 'ambassador') {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/common/support_view_static.js]: 👍

File: lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    this._renderBreadcrumbsDropdownLink();
+    this._renderSupportLink();
+    this._renderNotifications();
+    this._renderLogoLink();
+
+    return this;
+  },
+
+  _initBinds: function () {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js]: 👍

File: lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    }
+    if (this.collection) {
+      this.collection.bind('reset', this._stopLogoAnimation, this);
+      this.collection.bind('error', this._onCollectionError, this);
+      this.add_related_model(this.collection);
+    }
+    this.add_related_model(this._viewModel);
+  },
+
+  _onCollectionError: function (col, e, opts) {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js]: 👍

File: lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    // Old requests can be stopped, so aborted requests are not
+    // considered as an error
+    if (!e || (e && e.statusText !== 'abort')) {
+      this._stopLogoAnimation();
+    }
+  },
+
+  _onRouterChange: function (m, c) {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js]: it's actually `(model, event)` http://backbonejs.org/#Events-catalog

👍

File: lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    this.killEvent(ev);
+
+    this._setupDropdown(new SettingsDropdown({
+      target: $(ev.target),
+      model: this.model, // a user model
+      horizontal_offset: 18
+    }));
+  },
+
+  _createBreadcrumbsDropdown: function (ev) {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/common/views/dashboard_header_view_static.js]: 👍

File: lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+    this._initViews();
+    this._initBindings();
+  },
+
+  _initBindings: function () {
+    this.router.model.bind('change', this._onRouterChange, this);
+    this.collection.bind('reset', this._onDataFetched, this);
+    this.collection.bind('loading', this._onDataLoading, this);
+    this.collection.bind('add', this._onDataChange, this);
+    this.collection.bind('error', function (col, e, opts) {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js]: :thu

File: lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+    });
+
+    this.controlledViews['content_footer'] = contentFooter;
+    // Move element to end of the parent, if not paginator will appear
+    // before other elements
+    this.$('#content-footer').appendTo(this.$el);
+    this.addView(contentFooter);
+  },
+
+  _onRouterChange: function (m, c) {
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js]: it's actually `(model, value)` http://backbonejs.org/#Events-catalog

👍

File: lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js (hunk 1 above)
Comment (matallo) [reply lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js]: 👍

File: lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js (hunk 2 above)
Comment (matallo) [reply lib/assets/javascripts/cartodb/dashboard/content_controller_view_static.js]: 👍

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/imports_view.js [hunk 11]
```diff
@@ -80,10 +80,10 @@ module.exports = cdb.core.View.extend({
     var tabTemplate = cdb.templates.getTemplate('common/views/create/listing/import_tab');
 
     _.each(this._IMPORT_OPTIONS, function(t) {
-      if (!_.isEmpty(t) && t.enabled(cdb.config, this.user_data)) {
+      if (!_.isEmpty(t) && t.enabled(cdb.config, this.user)) {
         tabs += tabTemplate(t);
       }
-    });
+    }.bind(this));
```
[the remark this answers is not in the document]
Comment (matallo) [reply lib/assets/javascripts/cartodb/common/dialogs/create/listing/imports_view.js]: it's `tab`

👍

File: lib/assets/javascripts/cartodb/dashboard/main_view_static.js (hunk 3 above)
Comment (matallo) [reply lib/assets/javascripts/cartodb/dashboard/main_view_static.js]: 👍

**Pull request merged**